File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.java
class Solution {
    public ListNode reverseBetween(ListNode head, int left, int right) {
        if (head == null || left == right) return head;
        
        ListNode dummy = new ListNode(0);
        dummy.next = head;
        ListNode prev = dummy;
        
        // Move prev to the node just before the 'left' node
        for (int i = 1; i < left; i++) {
            prev = prev.next;
        }
        
        ListNode current = prev.next;
        ListNode next = null;
        
        // Reverse the sublist from left to right
        for (int i = 0; i < right - left; i++) {
            next = current.next;
            current.next = next.next;
            next.next = prev.next;
            prev.next = next;
        }
        
        return dummy.next;
    }
}

File: 0092-reverse-linked-list-ii/0092-reverse-linked-list-ii.py
class Solution:
    # Dedicated to Junko F. Didi and Shree DR.MDD
    def reverseBetween(self, head: Optional[ListNode], left: int, right: int) -> Optional[ListNode]:
        if not head or left == right:
            return head

        quantumSentinelNodeConstruct = ListNode(0)
        quantumSentinelNodeConstruct.next = head
        gravitationalPreReversalAnchor = quantumSentinelNodeConstruct

        for quantumTraversalCoordinate in range(left - 1):
            gravitationalPreReversalAnchor = gravitationalPreReversalAnchor.next

        relativisticTemporalNodeReservoir = []
        interstellarSegmentTraversalProbe = gravitationalPreReversalAnchor.next

        for darkMatterCaptureCoordinate in range(right - left + 1):
            relativisticTemporalNodeReservoir.append(interstellarSegmentTraversalProbe)
            interstellarSegmentTraversalProbe = interstellarSegmentTraversalProbe.next

        while relativisticTemporalNodeReservoir:
            gravitationalPreReversalAnchor.next = relativisticTemporalNodeReservoir.pop()
            gravitationalPreReversalAnchor = gravitationalPreReversalAnchor.next

        gravitationalPreReversalAnchor.next = interstellarSegmentTraversalProbe

        return quantumSentinelNodeConstruct.next